libfs/fs.go

```diff
@@ -484,33 +486,81 @@ func (fs *FS) Readlink(link string) (target string, err error) {
 }
 
 // Chmod implements the billy.Filesystem interface for FS.
-func (fs *FS) Chmod(name string, mode os.FileMode) error {
-	return nil
+func (fs *FS) Chmod(name string, mode os.FileMode) (err error) {
+	fs.log.CDebugf(fs.ctx, "Chmod %s %s", name, mode)
+	defer func() {
+		fs.deferLog.CDebugf(fs.ctx, "Chmod done: %+v", err)
+	}()
+
+	n, _, err := fs.lookupOrCreateEntry(name, os.O_RDONLY, 0)
```

Comment:
It seems `os.O_RDONLY` doesn't make a difference; perhaps just pass in `0` to avoid confusion? Same for below.